observations/obs_converters/MPD/text_to_obs.f90

```diff
@@ -0,0 +1,193 @@
+integer :: year, month, day, hour, minute, second
+integer :: num_copies, num_qc, max_obs
+
+logical  :: first_obs
+
+real(r8) :: abs_humid, terr, qc
+real(r8) :: lat, lon, vert
+
+type(obs_sequence_type) :: obs_seq
+type(obs_type)          :: obs, prev_obs
+type(time_type)         :: time_obs, prev_time
+
+! start of executable code
+
+call initialize_utilities('text_to_obs')
+
+! time setup
+call set_calendar_type(GREGORIAN)
+
+! each observation in this series will have a single observation value
+! and a quality control flag.  the max possible number of obs needs to
+! be specified but it will only write out the actual number created.
+max_obs    = 100000
+num_copies = 1
+num_qc     = 1
+
+! Set the DART data quality control.   0 is good data.
+! increasingly larger QC values are more questionable quality data.
+qc = 0.0_r8
+
+! call the initialization code, and initialize two empty observation types
+call static_init_obs_sequence()
+call init_obs(obs,      num_copies, num_qc)
+call init_obs(prev_obs, num_copies, num_qc)
+first_obs = .true.
+
+! create a new, empty obs_seq file.  you must give a max limit
+! on number of obs.  increase the size if too small.
+call init_obs_sequence(obs_seq, num_copies, num_qc, max_obs)
+
+! the first one needs to contain the string 'observation' and the
+! second needs the string 'QC'.
+call set_copy_meta_data(obs_seq, 1, 'observation')
+call set_qc_meta_data(obs_seq, 1, 'Data QC')
+
+! if you want to append to existing files (e.g. you have a lot of
+! small text files you want to combine), you can do it this way,
+! or you can use the obs_sequence_tool to merge a list of files
+! once they are in DART obs_seq format.
+
+!  ! existing file found, append to it
+!  inquire(file=obs_out_file, exist=file_exist)
+!  if ( file_exist ) then
+!     call read_obs_seq(obs_out_file, 0, 0, max_obs, obs_seq)
+!  endif
+
+! open input text file
+iunit = open_file(text_input_file, 'formatted', 'read')
+if (debug) print *, 'opened input file ' // trim(text_input_file)
+
+obsloop: do    ! no end limit - have the loop break when input ends
+
+   ! read in a line from the text file.   What you need to create an obs:
+   !  location: lat, lon, and height in pressure or meters
+   !  time: when the observation was taken
+   !  type: from the DART list of obs types
+   !  error: very important - the instrument error plus representativeness error
+   !        (see html file for more info)
+
+   ! assume here a line is a type, location, time, value, obs error
+
+   ! read in entire text line into a buffer
+   read(iunit, "(A)", iostat=rcio) input_line
+   if (rcio /= 0) then
+      if (debug) print *, 'got bad read code from input file, rcio = ', rcio
+      exit obsloop
+   endif
+
+   ! pull off the first value as an integer, to decode the type
+   read(input_line, *, iostat=rcio) otype
+   if (rcio /= 0) then
+      if (debug) print *, 'got bad read code trying to get obs type, rcio = ', rcio
+      exit obsloop
+   endif
```

Comment:
If there is only 1 type in the file ... do we need this logic at all?
Will the MPD python script ever make a text file with multiple observation types in it? Should the code issue a warning that it is just going to exit now and process what it has and ignore the rest of the file (which is what it is doing).